include/seqan3/range/detail/inherited_iterator_base.hpp

```diff
@@ -294,16 +305,34 @@ class inherited_iterator_base : public std::conditional_t<std::is_pointer_v<base
      * \{
     */
     //!\brief Dereference operator returns element currently pointed at.
-    constexpr reference operator*() const noexcept(noexcept(*std::declval<base_t &>()))
+    constexpr reference operator*() noexcept(noexcept(*std::declval<base_t &>()))
     //!\cond
-        requires std::InputIterator<base_t>
+        requires std::Readable<base_t>
+    //!\endcond
+    {
+        return **this_to_base();
+    }
+
+    //!\brief Dereference operator returns element currently pointed at.
+    constexpr decltype(auto) operator*() const noexcept(noexcept(*std::declval<base_t const &>()))
+    //!\cond
+        requires std::Readable<base_t>
     //!\endcond
     {
         return **this_to_base();
     }
 
     //!\brief Return pointer to this iterator.
-    constexpr pointer operator->() const noexcept(noexcept(*std::declval<base_t &>()))
+    constexpr pointer operator->() noexcept(noexcept(*std::declval<base_t &>()))
+    //!\cond
+        requires std::InputIterator<base_t>
+    //!\endcond
+    {
+        return &*this_to_base();
```

Comment:
don't get me wrong, but shouldn't it be `&**this_to_base()` 🤔
I mean you want to return a pointer to the referenced element not to the iterator itself. Why is the compiler not complaining? My head is going 💥

Reply:
the arrow operator induces itself recursively :sunglasses: 

> If a user-defined operator-> is provided, the operator-> is called again on the value that it returns, recursively, until an operator-> is reached that returns a plain pointer. After that, built-in semantics are applied to that pointer.

Reply:
wow, I didn't know